tests/intmap-properties.hs

```diff
@@ -922,6 +977,52 @@ prop_alter t k = case lookup k t of
     f Nothing   = Just ()
     f (Just ()) = Nothing
 
+#if MIN_VERSION_base(4,8,0)
+-- Verify that the rewrite rules for Identity give the same result as the
+-- non-rewritten version. We use our own TestIdentity functor to compare
+-- against.
+
+newtype TestIdentity a = TestIdentity { runTestIdentity :: a }
+
+instance Functor TestIdentity where
+    fmap = coerce
+
+prop_alterF_IdentityRules :: UMap -> Int -> Bool
+prop_alterF_IdentityRules t k =
+    runIdentity tIdentity == runTestIdentity tTestIdentity
```

Comment:
This should definitely check that the result is valid and should produce a `Property`.

Reply:
Done.